stubs/pyOpenSSL/OpenSSL/SSL.pyi

```diff
@@ -115,25 +117,63 @@ def SSLeay_version(type: int) -> str: ...
 class Session: ...
 
 class Connection:
-    def __getattr__(self, name: str) -> Any: ...  # incomplete
+    def __getattr__(self, name: str) -> Any: ...  # takes attributes from `self._socket`
     def __init__(self, context: Context, socket: socket.socket | None = ...) -> None: ...
+    def get_context(self) -> Context: ...
+    def set_context(self, context: Context) -> None: ...
+    def get_servername(self) -> bytes | None: ...
+    def set_tlsext_host_name(self, name: bytes) -> None: ...
+    def pending(self) -> int: ...
+    def send(self, buf: ReadableBuffer, flags: int = ...) -> int: ...
+    write = send
+    def sendall(self, buf: ReadableBuffer, flags: int = ...) -> int: ...
+    def recv(self, bufsiz: int, flags: int | None = ...) -> bytes: ...
+    read = recv
+    def recv_into(self, buffer: bytearray, nbytes: int | None = ..., flags: int | None = ...) -> int: ...
```

Comment:
Does `buffer` here have to be a bytearray, or can any writeable, mutable buffer do?

Reply:
It has to support:
- `len()`
- ```python
        # This strange line is all to avoid a memory copy. The buffer protocol
        # should allow us to assign a CFFI buffer to the LHS of this line, but
        # on CPython 3.3+ that segfaults. As a workaround, we can temporarily
        # wrap it in a memoryview.
        buffer[:result] = memoryview(_ffi.buffer(buf, result))
   ```

For example, `list` will also work:

```python
>>> l = []
>>> l[:4] = memoryview(b'abcd')
>>> l
[97, 98, 99, 100]
```

Reply:
```suggestion
    def recv_into(self, buffer: Sequence[int], nbytes: int | None = ..., flags: int | None = ...) -> int: ...
```

Reply:
`Sequence` does not support `[] =`. `MutableSequence` does:

```python
from typing import Sequence, MutableSequence

s: Sequence[int]
m: MutableSequence[int]

s[:1] = memoryview(b'1')  # Unsupported target for indexed assignment ("Sequence[int]")
m[:1] = memoryview(b'1')
```

Reply:
I see, the buffer is what is being assigned into. Technically `MutableSequence` isn't enough, because a mutable sequence could be indexable but not sliceable:

```
>>> d = collections.deque('asd')
>>> d[1] = 'hi'
>>> d[1:] = 'lol'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence index must be integer, not 'slice'
>>> 
```

I think `bytearray` and `MutableSequence` are both fine in practice. This method is mostly for compatibility with non-SSL sockets anyway, and doesn't perform any better than `.recv()` like it can for non-SSL sockets.

Reply:
`deque` is also a bit of a special case here — most classes that claim to be `MutableSequence`s should really support slicing